src/main/java/org/snomed/snowstorm/core/data/services/CodeSystemService.java

```diff
@@ -622,4 +642,16 @@ public void updateCodeSystemBranchMetadata(CodeSystem codeSystem) {
 		}
 		branchService.updateMetadata(branchPath, branchMetadata);
 	}
+
+	private String epochToImportDateMinusTime(long epoch, long minusTime) {
+		Date date = new Date(epoch);
+		date.setTime(date.getTime() - 5000);
+		return new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss").format(date);
+	}
+
+	private String epochToImportDatePlusTime(long epoch, long plusTime) {
+		Date date = new Date(epoch);
+		date.setTime(date.getTime() + 5000);
+		return new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss").format(date);
+	}
```

Comment:
@CoderMChu, I'm not 100% on this inflation (+/- 5 seconds) logic going to develop. Without it, a couple of tests will occasionally fail however. 

I'm not sure whether the failing tests themselves are at fault or they've helped expose a slight gap in our logic (there are a couple seconds overlay in versioning vs the heads of branches).

Reply:
Scenario A
- version has an import time of 09:00:03
- main has a head time of 09:00:03
- task has a head time of 09:00:01

In this scenario, without the inflation, the tests pass.

Scenario B
- version has an import time of 09:00:05
- main has a head time of 09:00:03
- task has a head time of 09:00:01

In this scenario, without the inflation, the test fails. That is why I have added a +/- 5 seconds.

Reply:
In theory, there shouldn't be a CodeSystem with multiple versions within this inflation... it just seems messy.